### PR TITLE
Update infrastructure

### PR DESCRIPTION
**Pull request by dantleech**
This PR:

- Switches to using autoloading
- Introduce a `phpcsfixer` configuration (and fix the CS)
- Removes the numerical prefix from the test directory names (`s/[0-9]{2}_//g`)

### CODE REVIEW

**Comment by dantleech** [comment]
Have executed the Jackalope-Doctrine-Dbal test suite and it works.

File: .php_cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+$header = <<<EOF
+This file is part of the PHPCR API Tests package
+
+Copyright (c) 2013 Liip and others
```
Comment (dantleech) [inline .php_cs]: 2015

File: .php_cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Symfony\CS\Fixer\Contrib\HeaderCommentFixer::setHeader($header);
+
+return Symfony\CS\Config\Config::create()
+    ->fixers(array(
+        'header_comment',
+        '-psr0',
+        'psr4',
+        'symfony',
+        'concat_with_spaces',
```
Comment (dantleech) [inline .php_cs]: Remove

File: tests/Import/ImportRepositoryContentTest.php [hunk 3]
```diff
@@ -374,7 +384,7 @@ public function testImportXMLDocumentSimple()
         self::$staticSharedFixture['ie']->import('11_Import/empty');
         $session = $this->renewSession();
 
-        $session->importXML('/', __DIR__.'/../../fixtures/11_Import/simple.xml', ImportUUIDBehaviorInterface::IMPORT_UUID_COLLISION_THROW);
+        $session->importXML('/', __DIR__ . '/../../fixtures/11_Import/simple.xml', ImportUUIDBehaviorInterface::IMPORT_UUID_COLLISION_THROW);
```
Comment (dantleech) [inline tests/Import/ImportRepositoryContentTest.php]: Hmm. Spaces or no spaces around `.` ?

**Comment by dantleech** [comment]
Reverted from `lib` backto `inc` to preserve BC.

File: tests/Import/ImportRepositoryContentTest.php (hunk 3 above)
Comment (dbu) [inline tests/Import/ImportRepositoryContentTest.php]: symfony cs would say no spaces afaik

**Comment by lsmith77** [comment]
looks good to me!

**Comment by lsmith77** [comment]
I guess since we use explicit tagged versions in Jackalope, this should also not cause breakage in older branches .. right?

**Comment by dantleech** [comment]
Yeah, I guess if we have a policy of using explicit versions for phpcr-api-tests in jackalope, it would be safe. But should not break anything even in older versions atm, as its fully BC (tm).

File: .php_cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+$header = <<<EOF
+This file is part of the PHPCR API Tests package
+
+Copyright (c) 2015 Liip and others
+
+For the full copyright and license information, please view the LICENSE
+file that was distributed with this source code.
+EOF;
+
+Symfony\CS\Fixer\Contrib\HeaderCommentFixer::setHeader($header);
```
Comment (dbu) [inline .php_cs]: when / how is this triggered?

**Comment by dbu** [comment]
cool, makes sense. i would love to keep the chapter numbers in the namespaces however. can a php namespace start with a number? then we could namespace them as 06Query or something.

File: .php_cs (hunk 4 above)
Comment (dantleech) [inline .php_cs]: It would need to be executed manually: `/path/to/phpcsfixer fix`

**Comment by dantleech** [comment]
No, it seems [namespaces cannot start with numbers](http://php.net/manual/en/language.namespaces.rationale.php#104196). I would suggest to add the chapter information to the class documentation, and also the relevant parts of the spec to each test method doc (if this is not already the case).

**Comment by lsmith77** [comment]
sounds good to me ..

File: .php_cs (hunk 4 above)
Comment (dbu) [inline .php_cs]: we should have that in a doc comment then.

**Comment by dbu** [comment]
well, it was very useful to find tests when following the specification. but i guess i will just have to change my habits. the benefits of this PR are worth more than this convenience :-)

the spec is not copied into the tests. sometimes the spec subsections covered are mentioned, but not consistently. while that certainly would be great, it sounds like a huge work. lets document the cs fix thingy a bit and then merge this.

**Comment by dbu** [comment]
@dantleech thanks! now we need to adjust the jackalopes...

**Pull request merged**